src/components/BShuttle/BShuttle.vue

```diff
@@ -0,0 +1,480 @@
+          @keyup.left.prevent.stop="onMoveSelectedClick('chosen')"
+          @keyup.enter.prevent.stop="onMoveSelectedClick('chosen')"
+          @keydown.up.prevent.stop
+          @keydown.down.prevent.stop
+          @keydown.space.prevent.stop
+          role='option'
+          tabindex='0'
+        >
+          {{ option.displayText }} 
+
+        </li>
+      <span v-show="errors.has(validationId)" class="error-text" :id="'error-' + id">{{ errors.first(validationId) }}</span>
+      </ul>
+
+    </div>
+
+  </div>
+  
+</template>
+
+<script>
+
+import validationMixIn from '../../mixins/validation'
+
+export default {
+  name: 'b-shuttle',
+  mixins: [validationMixIn()],
+  props: {
+    /**
+    * The available options for the shuttle.
+    */
+    options: {
+      type: Array,
+      required: false
+    },
+    /**
+    * Placeholder to display if shuttle is dependent on other component.
+    */
+    placeholder: {
+      type: String,
+      required: false
+    },
+    /**
+    * Label to display above available options.
+    */
+    availableLabel: {
+      type: String,
+      required: true
+    },
+    /**
+    * Label to display above chosen options.
+    */
+    chosenLabel: {
+      type: String,
+      required: true
+    },
+    /**
+    * Use to turn on validation
+    */
+    required: {
+      type: Boolean,
+    },
+    /**
+    * The id(s) of the selected option(s).
+    */
+    value: {
+      type: Array,
+      required: false,
+      default: function () { return [] }
+    }
+  },
+
+  data () {
+    return {
+      availableOptions: (this.options && this.options.length > 0) ? this.setOptions(this.options, this.value, 'available') : new Array,
+      selectedOptions: {'available': new Array, 'chosen': new Array},
+      chosenOptions: (this.value && this.options) ? this.setOptions(this.options, this.value, 'chosen') : new Array,
+      lastClick: 0,
+      id: null,
+      allToChosen: require('../../../static/doubleArrow.svg'),
+      allToAvailable: require('../../../static/doubleArrow-left.svg')
+    }
+  },
+
+  computed: {
+  },
+
+  created() {
+    this.$validator.extend('notEmpty', {
+      getMessage: field => 'The ' + field + ' input is required.',
+      validate: value => value.length > 0
+    });
+    this.$validator.attach({
+      name: this.validationId,
+      rules: 'notEmpty',
+      alias: this.chosenLabel
+    });
+
+  },
+
+  mounted() {
+    this.id = this._uid
+  },
+
+  methods: {
+
+    /**
+     * @param {Array<Object>} options
+     * @param {Array<Number>} selected
+     * @param {String} type
+     * this method sets up the initial state of the options
+     */
+    setOptions: function (options, selected, type) {
+      const availableOptions = new Array
+      const chosenOptions = new Array
+      options.forEach(function (option) {
+        if (selected === null) {
+          availableOptions.push(option)
+        } else if (selected.indexOf(option.id) === -1) {
```

Comment:
Making this an `||` since they have the same outcome

Reply:
👍